Code/GraphMol/CIPLabeler/CIPLabeler.cpp

```diff
@@ -0,0 +1,203 @@
+            if (atom == nbr_bnd->getEndAtom()) {
+              if (nbr_bond_dir == Bond::BondDir::ENDDOWNRIGHT) {
+                nbr_bond_dir = Bond::BondDir::ENDUPRIGHT;
+              } else {
+                nbr_bond_dir = Bond::BondDir::ENDDOWNRIGHT;
+              }
+            }
+
+            bond_cfg += nbr_bond_dir == previous_bond_dir;
+            std::swap(previous_bond_dir, nbr_bond_dir);
+            break;
+          }
+        }
+      }
+
+      if (anchors.size() == 2) {
+        auto cfg =
+            new Sp2Bond(bond, std::move(atoms), std::move(anchors), bond_cfg);
+        configs.push_back(cfg);
+      }
+    }
+  }
+
+  return configs;
+}
+
+bool labelAux(const std::vector<Configuration *> &configs, const Rules *rules,
+              const Configuration *center) {
+  using Node_Cfg_Pair = std::pair<Node *, Configuration *>;
+  auto aux = std::vector<Node_Cfg_Pair>{};
+
+  auto digraph = center->getDigraph();
+  for (const auto &config : configs) {
+    if (config == center) {
+      continue;
+    }
+    // FIXME: specific to each descriptor
+    const auto &foci = config->getFoci();
+    for (const auto &node : digraph->getNodes(foci[0])) {
+      if (node->isDuplicate()) {
+        continue;
+      }
+      auto low = node;
+      if (foci.size() == 2) {
+        for (const auto &edge : node->getEdges(foci[1])) {
+          if (edge->getOther(node)->getDistance() < node->getDistance())
+            low = edge->getOther(node);
+        }
+      }
+      if (!low->isDuplicate()) {
+        aux.emplace_back(low, config);
+      }
+    }
+  }
+
+  auto pair_cmp = [](const Node_Cfg_Pair &a, const Node_Cfg_Pair &b) {
+    return a.first->getDistance() > b.first->getDistance();
+  };
+  std::sort(aux.begin(), aux.end(), pair_cmp);
+
+  auto queue = boost::unordered_map<Node *, Descriptor>{};
+  int prev = std::numeric_limits<int>::max();
+  for (const auto &e : aux) {
+    const auto &node = e.first;
+
+    if (node->getDistance() < prev) {
+      for (const auto &e2 : queue) {
+        e2.first->setAux(e2.second);
+      }
+      queue.clear();
+      prev = node->getDistance();
+    }
+    const auto &config = e.second;
+    auto label = config->label(node, digraph.get(), rules);
+    queue.emplace(node, label);
+  }
+
+  for (const auto &e : queue) {
+    e.first->setAux(e.second);
+  }
+
+  return true;
+}
+
+void label(CIPMol *mol, const std::vector<Configuration *> &configs) {
+  // constitutional rules
+  const Rules begRules({new Rule1a(mol), new Rule1b(mol), new Rule2(mol)});
+
+  // all rules (require aux calc)
+  const Rules allRules({new Rule1a(mol), new Rule1b(mol), new Rule2(mol),
+                        new Rule3(mol), new Rule4a(mol), new Rule4b(mol),
+                        new Rule4c(mol), new Rule5New(mol), new Rule6(mol)});
+
+  auto finalLabels = boost::unordered_map<Configuration *, Descriptor>{};
+  for (const auto &conf : configs) {
+    conf->setDigraph(new Digraph(mol));
```

Comment:
I don't think this is possible in all cases: as the digraphs are expanded node by node when resolving one stereo center, I think it might be quite common to find that the nodes matching other stereo centers have not even been added to the digraph when the first one has been resolved.